src/Core/Data/BoundedDeque.hpp

```
@@ -0,0 +1,136 @@
+		if (m_deque.size() >= Capacity)
+			return false;
+
+		pushFunc();
+		m_dequeNotEmpty.notify_one();
+	
+		return true;
+	}
+
+
+	template<typename GetFunc, typename PopFunc>
+	T pop(GetFunc&& getFunc, PopFunc&& popFunc) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)